frontend/src/ui/Access.tsx

```diff
@@ -0,0 +1,698 @@
+                    ref={groupsRef}
+                    kind="Group"
+                    initialAcl={initialGroupAcl}
+                    allOptions={groupOptions}
+                />
+                <AclSelect
+                    ref={usersRef}
+                    kind="User"
+                    initialAcl={initialUserAcl}
+                    allOptions={userOptions}
+                />
+            </div>
+            {children}
+        </AclContext.Provider>;
+    }
+);
+
+
+type AclSelectProps = SelectProps & {
+    initialAcl: Acl;
+    allOptions: Option[];
+    kind: "Group" | "User";
+};
+
+type AclSelectHandle = {
+    getSelection: () => MultiValue<Option>;
+    reset: () => void;
+};
+
+type SelectContext = {
+    item: Option;
+    kind: "User" | "Group";
+};
+
+const defaultDummyOption: Option = {
+    value: COMMON_ROLES.ADMIN,
+    label: "Administrator",
+};
+
+const SelectContext = createContext<SelectContext>({
+    item: defaultDummyOption,
+    kind: "User",
+});
+
+const AclSelect = forwardRef<AclSelectHandle, AclSelectProps>(
+    ({ initialAcl, allOptions, kind }, ref) => {
+        const isDark = useColorScheme().scheme === "dark";
+        const user = useUser();
+        const { t } = useTranslation();
+        const { roleSelections, setRoleSelections } = useContext(AclContext);
+        const [menuIsOpen, setMenuIsOpen] = useState<boolean>(false);
+
+        const translations = match(kind, {
+            "Group": () => ({
+                heading: t("manage.access.authorized-groups"),
+                placeholder: t("manage.access.select.groups"),
+                columnHeader: t("manage.access.table.group"),
+            }),
+            "User": () => ({
+                heading: t("manage.access.authorized-users"),
+                placeholder: t("manage.access.select.users"),
+                columnHeader: t("manage.access.table.user"),
+            }),
+        });
+
+        const isSubset = (role: string, potentialSuperset: string): boolean => {
+            const relation = SUBSET_RELATIONS.find(entry => entry.superset === potentialSuperset);
+            if (relation) {
+                return relation.subsets.includes(role)
+                    || relation.subsets.some(subset => isSubset(role, subset));
+            }
+            return false;
+        };
+
+        const roleComparator = (a: Option, b: Option) => {
+            if (kind === "Group") {
+                // Sort ACL group entries by their scope,
+                // so that supersets will be shown before subsets.
+
+                // A is a subset of b, so b should come first.
+                if (isSubset(a.value, b.value)) {
+                    return 1;
+                }
+                // B is a subset of a, so a should come first.
+                if (isSubset(b.value, a.value)) {
+                    return -1;
+                }
+                // Neither is a subset of the other, don't sort.
+                return 0;
```

Comment:
This does not satisfy the "transitive" property that's required (see [docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#description)):

> More formally, the comparator is expected to have the following properties, in order to ensure proper sort behavior:
>
> [...]
> - *Transitive*: If `compareFn(a, b)` and `compareFn(b, c)` are both positive, zero, or negative, then `compareFn(a, c)` has the same positivity as the previous two.

Assuming that `isSubset` implements a correct strict set subset check, then the following inputs fail transitivity:

- a: { 🐑 🐭 }
- b: { 🐔 }
- c: { 🐑 }

`compareFn(a, b)` and `compareFn(b, c)` both return 0 as there is no subset relationship. But `compareFn(a, c)` returns non-0 as there is a subset.

In that case, the behavior of `sort` is "implementation defined": https://stackoverflow.com/a/40902509/2408867
Which is not great as we in theory cannot expect any reasonable output. In fact, we cannot even expect that subsets appear after their supersets. That said, the worst that could happen is a weird sort order of the entries, which is not a catastrophe. And I strongly suspect that the result will always be roughly what we expect. So maybe we can just keep it as is?

How could we even fix it? I don't think we can't use `sort`. Sort simply requires a total ordering, meaning that any two elements must be comparable. But in our case, { 🐑 🐭 } and { 🐔 } cannot be compared! Your code returns 0 (meaning: equal) here, but that's not true of course. We simply cannot establish an ordering between the two elements.

So I think we need to use a topological sort, as that's the right tool for the job. Whether to implement that ourselves (I would :D) or use a library is up to you. Though we shouldn't pull in a super heavy general graph processing library or something like that.